pip/cmdoptions.py

```diff
@@ -523,6 +524,58 @@ def only_binary():
 )
 
 
+def _good_hashes():
+    """Return names of hashlib algorithms at least as strong as sha256."""
+    # Remove getattr when 2.6 dies.
+    algos = set(
+        getattr(hashlib,
+                'algorithms',
+                ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')))
+    return algos - set(['md5', 'sha1', 'sha224'])
```

Comment:
Unfortunately, I don't think digest size is an authoritative indicator of hash strength. I think we have to go with either a whitelist or a blacklist. I chose a blacklist so people would have the ability to use newer, stronger (or even custom) hashes without waiting for a pip update. Would you rather play it safe and go with a whitelist?